test/e2e/framework/framework.go

```diff
@@ -206,16 +117,68 @@ func KubernetesClientEnv(t *testing.T) []string {
 	return []string{"KUBERNETES_SERVICE_HOST=" + host, "KUBERNETES_SERVICE_PORT=" + port}
 }
 
+type Accessory interface {
+	Ready()
+	Bind(cmd CiOperatorCommand)
+}
+
+var _ Accessory = &accessory{}
+
+type accessory struct {
+	command    string
+	port       string
+	healthPort string
+
+	binder func(port string, cmd CiOperatorCommand)
+
+	t T
+}
+
+func (a *accessory) Ready() {
+	if waitErr := wait.PollImmediate(1*time.Second, 30*time.Second, func() (done bool, err error) {
+		resp, getErr := http.Get(fmt.Sprintf("http://127.0.0.1:%s/healthz/ready", a.healthPort))
+		defer func() {
+			if resp == nil || resp.Body == nil {
+				return
+			}
+			if closeErr := resp.Body.Close(); closeErr != nil {
+				a.t.Logf("could not close response body: %v", closeErr)
+			}
+		}()
+		if resp != nil {
+			a.t.Logf("`%s` readiness probe: %v", a.command, resp.StatusCode)
+		}
+		if getErr != nil {
+			a.t.Logf("`%s` readiness probe error: %v:", a.command, getErr)
+		}
+		return (resp != nil && resp.StatusCode == http.StatusOK) && getErr == nil, nil
+	}); waitErr != nil {
+		a.t.Fatalf("could not wait for `%s` to be healthy: %v", a.command, waitErr)
+	}
+}
+
+func (a *accessory) Bind(cmd CiOperatorCommand) {
```

Comment:
I find the `accessory` abstraction extremely confusing, why is this modeled in a way where we do the running in a different level than the readyness checking? It would be much more logical to start the binary in a `Run` method of the `accessory` rather than in an accessory constructor

Reply:
I guess I'm a proponent of not letting people do the wrong thing unless they really mean it. I don't want to expose this since you can get into really hairy, irritating deadlocks and timeouts if you mess with things ...